kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/ssl/KubernetesClientSslBuilder.java

```diff
@@ -0,0 +1,120 @@
+
+import java.io.ByteArrayInputStream;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * The ssl builder which uses data from a kube config file to create a key store and trust store.
+ */
+@Internal
+public class KubernetesClientSslBuilder extends NettyClientSslBuilder {
+
+    private KubeConfig kubeConfig;
+    private KubernetesPrivateKeyLoader kubernetesPrivateKeyLoader;
+
+    public KubernetesClientSslBuilder(ResourceResolver resourceResolver,
+                                      KubeConfig kubeConfig,
+                                      KubernetesPrivateKeyLoader kubernetesPrivateKeyLoader) {
+        super(resourceResolver);
+        this.kubeConfig = kubeConfig;
+        this.kubernetesPrivateKeyLoader = kubernetesPrivateKeyLoader;
+    }
+
+    @Override
+    protected Optional<KeyStore> getKeyStore(SslConfiguration ssl) throws Exception {
+        AuthInfo user = kubeConfig.getUser();
+        byte[] clientCert = user.clientCertificateData();
+        byte[] clientKey = user.clientKeyData();
+        if (clientCert == null || clientKey == null) {
+            return Optional.empty();
+        }
+
+        CertificateFactory certFactory = CertificateFactory.getInstance("X509");
+        Collection<? extends Certificate> certs = certFactory.generateCertificates(new ByteArrayInputStream(clientCert));
+
+        String keyAlias;
+        if (ssl.getKey().getAlias().isPresent()) {
+            keyAlias = ssl.getKey().getAlias().get();
+        } else {
+            keyAlias = ((X509Certificate) certs.stream().findFirst().get()).getSubjectX500Principal().getName();
+        }
+
+        PrivateKey privateKey = kubernetesPrivateKeyLoader.loadPrivateKey(clientKey);
+
+        String keyPass;
+        if (ssl.getKey().getPassword().isPresent()) {
+            keyPass = ssl.getKey().getPassword().get();
+        } else if (ssl.getKeyStore().getPassword().isPresent()) {
+            keyPass = ssl.getKeyStore().getPassword().get();
+        } else {
+            keyPass = "";
+            ssl.getKey().setPassword("");
+        }
+
+        KeyStore keyStore = KeyStore.getInstance(ssl.getKeyStore().getType().orElse("JKS"));
+        keyStore.load(null);
+        keyStore.setKeyEntry(keyAlias, privateKey, keyPass.toCharArray(), certs.toArray(new X509Certificate[0]));
+        return Optional.of(keyStore);
+    }
+
+    @Override
+    protected Optional<KeyStore> getTrustStore(SslConfiguration ssl) throws Exception {
+        Cluster cluster = kubeConfig.getCluster();
+
+        Boolean insecureSkipTlsVerify = cluster.insecureSkipTlsVerify();
+        if (insecureSkipTlsVerify != null && insecureSkipTlsVerify) {
+            ((ClientSslConfiguration) ssl).setInsecureTrustAllCertificates(true);
```

Comment:
I assume the default is not to do this?

Reply:
The value is read from the kube config. If it is not set in the kube config, the behaviour will be the same as it is set to `false`.